src/components/ModalWithForm.js

```diff
@@ -33,8 +32,14 @@ class ModalWithForm extends Modal {
     return inputValues;
   }
   setEventListeners() {
-    this._modalForm.addEventListener("submit", (event) => {
-      event.preventDefault();
+    this.modalSubmitButton = this._modalForm.querySelector(
+      ".modal__submit-button"
+    );
+
+    this._modalForm.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this.modalSubmitButton.textContent = "Saving...";
```

Comment:
Needs correction: Manage loading from `index.js`. It's important to start loading right before the api call and end loading in promise's `finally` block (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally), because loading should end even if an error occurred

One of the options to implement that would be adding a parameter `loadingButtonText` to the constructor of `PopupWithForm` and using it to change button text. 

```js
constructor(popupSelector, { handleSubmit, loadingButtonText }) { 
  ...
  this._submitButton = this._formElement.querySelector(...)
  this._buttonText = this._submitButton.textContent;
  this._loadingButtonText = loadingButtonText;
  …
}

showLoading() {
  this._submitButton.textContent = this._loadingButtonText
}

hideLoading() {
  this._submitButton.textContent = this._buttonText
}
``` 

The api call should look like this:

```js
popup.showLoading()
api.
  .then(...)
  .catch(...)
  .finally(() => {
    popup.hideLoading()

  })
```